lib/npm-installation-manager.ts

```diff
@@ -23,103 +13,51 @@ export class NpmInstallationManager implements INpmInstallationManager {
-				cachedPackageData = this.addCleanCopyToCache(packageName, version).wait();
-			}
-
-			return cachedPackageData;
-		}).future<any>()();
-	}
-
-	public cacheUnpack(packageName: string, version: string, unpackTarget?: string): IFuture<void> {
-		unpackTarget = unpackTarget || path.join(npm.cache, packageName, version, "package");
-		return this.$npm.cacheUnpack(packageName, version, unpackTarget);
+	public getLatestVersion(packageName: string): IFuture<string> {
+		return(() => {
+			return this.getVersion(packageName, constants.PackageVersion.LATEST);
+		}).future<string>()();
 	}
 
-	public getLatestVersion(packageName: string): IFuture<string> {
+	public getNextVersion(packageName: string): IFuture<string> {
 		return (() => {
-			let data = this.$npm.view(packageName, "dist-tags").wait();
-			// data is something like :
-			// { '1.0.1': { 'dist-tags': { latest: '1.0.1', next: '1.0.2-2016-02-25-181', next1: '1.0.2' } }
-			// There's only one key and it's always the @latest tag.
-			let latestVersion = _.first(_.keys(data));
-			this.$logger.trace("Using version %s. ", latestVersion);
-
-			return latestVersion;
+			return this.getVersion(packageName, constants.PackageVersion.NEXT);
 		}).future<string>()();
 	}
 
 	public getLatestCompatibleVersion(packageName: string): IFuture<string> {
 		return (() => {
+
 			let cliVersionRange = `~${this.$staticConfig.version}`;
 			let latestVersion = this.getLatestVersion(packageName).wait();
 			if(semver.satisfies(latestVersion, cliVersionRange)) {
 				return latestVersion;
 			}
+			let data: any = this.$npm.view(packageName, {json: true, "versions": true});
 
-			let data: any = this.$npm.view(packageName, "versions").wait();
-			/* data is something like:
-				{
-					"1.1.0":{
-						"versions":[
-							"1.0.0",
-							"1.0.1-2016-02-25-181",
-							"1.0.1",
-							"1.0.2-2016-02-25-182",
-							"1.0.2",
-							"1.1.0-2016-02-25-183",
-							"1.1.0",
-							"1.2.0-2016-02-25-184"
-						]
-					}
-				}
-			*/
-			let versions: string[] = data && data[latestVersion] && data[latestVersion].versions;
-			return semver.maxSatisfying(versions, cliVersionRange) || latestVersion;
+			return semver.maxSatisfying(data, cliVersionRange) || latestVersion;
 		}).future<string>()();
 	}
 
-	public install(packageName: string, opts?: INpmInstallOptions): IFuture<string> {
+	public install(packageName: string, projectDir: string, opts?: INpmInstallOptions): IFuture<any> {
 		return (() => {
 
+			// TODO: plamen5kov: figure a way to remove this
 			while(this.$lockfile.check().wait()) {
 				sleep(10);
 			}
 
 			this.$lockfile.lock().wait();
 
 			try {
-				let packageToInstall = packageName;
-				let pathToSave = (opts && opts.pathToSave) || npm.cache;
+				let packageToInstall = this.$options.frameworkPath || packageName;
+				let pathToSave = projectDir;
 				let version = (opts && opts.version) || null;
+				let dependencyType = (opts && opts.dependencyType) || null;
```

Comment:
dependency type is the npm flag that you want to use on install like --save --save-dev ...